e2e-polybft/consensus_test.go

```diff
@@ -174,28 +237,54 @@ func TestE2E_Consensus_RegisterValidator(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, extra.Checkpoint.NextValidatorsHash, validatorsHash)
 
-	// query registered validator
-	newValidatorInfo, err := sidechain.GetValidatorInfo(newValidatorAddr, txRelayer)
+	// query the first validator
+	firstValidatorInfo, err := sidechain.GetValidatorInfo(firstValidatorAddr, txRelayer)
+	require.NoError(t, err)
+
+	// assert the first validator's stake
+	t.Logf("First validator stake=%d\n", firstValidatorInfo.TotalStake)
+	require.Equal(t, newValidatorStake, firstValidatorInfo.TotalStake)
+
+	// query the second validatorr
+	secondValidatorInfo, err := sidechain.GetValidatorInfo(secondValidatorAddr, txRelayer)
 	require.NoError(t, err)
 
-	// assert registered validator's stake
-	t.Logf("New validator stake=%d\n", newValidatorInfo.TotalStake)
-	require.Equal(t, newValidatorStake, newValidatorInfo.TotalStake)
+	// assert the second validator's stake
+	t.Logf("Second validator stake=%d\n", secondValidatorInfo.TotalStake)
+	require.Equal(t, newValidatorStake, secondValidatorInfo.TotalStake)
 
 	// wait 3 more epochs, so that rewards get accumulated to the registered validator account
-	currentBlock, err := srv.JSONRPC().Eth().BlockNumber()
+	currentBlock, err := owner.JSONRPC().Eth().BlockNumber()
 	require.NoError(t, err)
 	cluster.WaitForBlock(currentBlock+3*epochSize, 2*time.Minute)
 
-	// query registered validator
-	newValidatorInfo, err = sidechain.GetValidatorInfo(newValidatorAddr, txRelayer)
+	// query the first validator info again
+	firstValidatorInfo, err = sidechain.GetValidatorInfo(firstValidatorAddr, txRelayer)
 	require.NoError(t, err)
 
-	// assert registered validator's rewards
-	currentBlock, err = srv.JSONRPC().Eth().BlockNumber()
+	// check if the first validator has signed any prposals
+	firstSealed, err := hasValidatorSealed(firstValidator, currentBlock, currentBlock+3*epochSize, validators, firstValidatorAddr)
 	require.NoError(t, err)
-	t.Logf("New validator rewards (block %d)=%d\n", currentBlock, newValidatorInfo.WithdrawableRewards)
-	require.True(t, newValidatorInfo.WithdrawableRewards.Cmp(big.NewInt(0)) > 0)
+
+	if firstSealed {
+		// assert registered validator's rewards)
+		t.Logf("First validator rewards (block %d)=%d\n", currentBlock, firstValidatorInfo.WithdrawableRewards)
```

Comment:
WDYT?
```suggestion
	} else {
		require.True(t, firstValidatorInfo.WithdrawableRewards.Cmp(big.NewInt(0)) == 0)
	}
```

Reply:
Statement under the `else` is not guaranteed to be `true`. Maybe the new validator received some rewards before or after the blocks we have checked.